Src/Base/AMReX_BLProfiler.H

```diff
@@ -484,10 +484,17 @@ inline std::string BLProfiler::CommStats::CFTToString(CommFuncType cft) {
 #define BL_PROFILE_S(fname)
 #define BL_PROFILE_T_S(fname, T)
 
-#define BL_PROFILE_VAR(fname, vname)      amrex::TinyProfiler tiny_profiler__##vname((fname));
-#define BL_PROFILE_VAR_NS(fname, vname)   amrex::TinyProfiler tiny_profiler__##vname(fname, false);
-#define BL_PROFILE_VAR_START(vname)       tiny_profiler__##vname.start();
-#define BL_PROFILE_VAR_STOP(vname)        tiny_profiler__##vname.stop();
+#define BL_PROFILE_VAR(fname, vname)                      amrex::TinyProfiler tiny_profiler__##vname((fname));
+#define BL_PROFILE_VAR_NS(fname, vname)                   amrex::TinyProfiler tiny_profiler__##vname(fname, false);
+#ifdef AMREX_USE_CUPTI
+#define BL_PROFILE_VAR_NS_CUPTI(fname, vname)             amrex::TinyProfiler tiny_profiler__##vname(fname, false, true);
+#endif
+#define BL_PROFILE_VAR_START(vname)                       tiny_profiler__##vname.start();
+#define BL_PROFILE_VAR_STOP(vname)                        tiny_profiler__##vname.stop();
+#ifdef AMREX_USE_CUPTI
+#define BL_PROFILE_VAR_START_CUPTI(vname)                 tiny_profiler__##vname.start();
+#define BL_PROFILE_VAR_STOP_CUPTI(vname, uintID, charID)  tiny_profiler__##vname.stop(uintID, charID);
+#endif
 #define BL_PROFILE_INIT_PARAMS(ptl,wall,wfabs)
 #define BL_PROFILE_ADD_STEP(snum)
```

Comment:
What happens to *_CUPTI macros when TINY_PROFILE is off? Looks like this is only the "on" situation.

Reply:
The macros are not defined when TINY_PROFILE is off, but added a CuptiTrace class that can be used for kernel timings even without TINY_PROFILE.